packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -569,12 +581,25 @@ export class ConnectionManager implements IConnectionManager {
 				lastError = origError;
 
 				const retryDelayFromError = getRetryDelayFromError(origError);
-				delayMs = retryDelayFromError ?? Math.min(delayMs * 2, MaxReconnectDelayInMs);
-
 				if (retryDelayFromError !== undefined) {
+					// If the error told us to wait, then we wait.
 					this.props.reconnectionDelayHandler(retryDelayFromError, origError);
+					await new Promise<void>((resolve) => {
```

Comment:
I think we have delay() function somewhere that does that.

Reply:
We do but it's in common-utils, which I expect I'll go rearrange relatively soon (need to figure out what its relationship should be relative to the new core-utils package).  For now I don't want to take new dependencies on it.